dvc/exceptions.py

```diff
@@ -338,3 +337,9 @@ def __init__(self, target, file):
 
 class MergeError(DvcException):
     pass
+
+
+class AuthInfoMissingError(DvcException):
+    """
+    Raised when not enough information is provided for auth against remote.
+    """
```

Comment:
Let's keep it in `dvc/tree/azure.py` for now, since it is only used there. I see that you've probably added it here for a potential to use it in the future in other remotes, which is good thinking :ok_hand: , but let's just keep it local for now. :slightly_smiling_face:

Reply:
So my approach of throwing an exception if insufficient info was present was causing an issue with the `dvc\tests\unit\dependency\test_azure.py` test due to the `TestLocalDependency` superclass to `TestAzureDependency`.

 `dvc\tests\unit\dependency\test_azure.py` -> test_save_missing()` was breaking. Drilling down, I noticed that an empty config object was being passed into the tree. 
I saw no way of injecting a appropriate config object, so instead, rather than throwing the exception, I simply write an error log with some message!

Reply:
So my approach of throwing an exception if insufficient info was present was causing an issue with the dvc\tests\unit\dependency\test_azure.py test due to the TestLocalDependency superclass to TestAzureDependency.

dvc\tests\unit\dependency\test_azure.py -> test_save_missing()` was breaking. Drilling down, I noticed that an empty config object was being passed into the tree.
I saw no way of injecting a appropriate config object, so instead, rather than throwing the exception, I simply write an error log with some message!